app/assets/src/components/views/PipelineViz.jsx

```diff
@@ -0,0 +1,366 @@
+
+  adjustStageWidth(graph) {
+    // Set initial zoom for width calculation.
+    graph.moveTo({ scale: 1 });
+
+    // Calculate and set new canvas width.
+    const minX = graph.canvasToDOM({
+      x: graph.getBoundingBox(START_NODE_ID).right,
+    }).x;
+    // Use right instead of left to include room for edge arrow tip.
+    const maxX = graph.canvasToDOM({
+      x: graph.getBoundingBox(END_NODE_ID).right,
+    }).x;
+    graph.setSize(maxX - minX + "px", "100%");
+
+    // Reset zoom (which is adjusted in setSize).
+    graph.moveTo({ scale: 1, position: { x: 0, y: 0 } });
+  }
+
+  renderGraphs() {
+    this.modifyStepNames();
+    this.stageNames.forEach((stageName, i) => {
+      this.renderStageGraph(
+        this.stageResults[stageName],
+        this[`container${i}`],
+        i
+      );
+    });
+    this.renderInterStageEdges();
+    this.adjustGraphNodePositions();
+  }
+
+  renderInterStageEdges() {
+    this.stageGraphsInfo.forEach((currStageInfo, i) => {
+      if (i == this.stageGraphsInfo.length - 1) {
+        // Create hidden edges to final node for vertical centering of nodes.
+        currStageInfo.data.nodes.forEach(nodeData => {
+          currStageInfo.data.edges.add({
+            from: nodeData.id,
+            to: END_NODE_ID,
+            color: {
+              color: STAGE_BG_COLOR,
+              inherit: false,
+            },
+            chosen: false,
+          });
+        });
+      } else {
+        const currStageData = this.stageResults[this.stageNames[i]];
+
+        const currFileNameToOutputtingNode = {};
+        currStageData.steps.forEach((step, nodeId) => {
+          currStageData.targets[step.out].forEach(fileName => {
+            fileName = `${currStageData.output_dir_s3}/${
+              this.pipelineVersion
+            }/${fileName}`;
+            currFileNameToOutputtingNode[fileName] = nodeId;
+          });
+        });
+
+        const nextStageData = this.stageResults[this.stageNames[i + 1]];
+        nextStageData.steps.forEach((step, nextNodeId) => {
+          step.in.forEach(inTarget => {
+            nextStageData.targets[inTarget].forEach(fileName => {
+              if (inTarget in nextStageData.given_targets) {
+                fileName = `${
+                  nextStageData.given_targets[inTarget].s3_dir
+                }/${fileName}`;
+                if (fileName in currFileNameToOutputtingNode) {
+                  const currNodeId = currFileNameToOutputtingNode[fileName];
+                  currStageInfo.data.edges.add({
+                    from: currNodeId,
+                    to: END_NODE_ID,
+                  });
+                  // TODO(ezhong): Interactions between output of current stage (currNodeId)
+                  // and input of next stage (nextNodeId) in visualization should be setup here
+                }
+              }
+            });
+          });
+        });
+      }
+    });
+  }
+
+  renderStageGraph(stageData, container, index) {
```

Comment:
since this function isn't rendering anything you could call it prepareStageGraph or similar